### PR TITLE
20419 - EFT Short name link additional info

### PR DESCRIPTION
**Pull request by ochiu**
*Issue #:*
https://github.com/bcgov/entity/issues/20419
*Description of changes:*
- extend BaseVDataTable to support expanded rows for additional information
- update ShortNameAccountLink component to make use of expanded rows to display additional information
- Add Unlink for pending links

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-web/src/components/pay/eft/ShortNameAccountLink.vue [hunk 1]
```diff
@@ -57,50 +58,79 @@
...
           <span>{{ formatCurrency(item.amountOwing) }}</span>
         </template>
-        <template #item-slot-actions="{ index }">
+        <template #expanded-item="{ item }">
+          <tr class="expanded-item-row">
+            <td
+              :colspan="headers.length"
+              class="pb-5 pl-0"
+            >
+              <span class="expanded-item scheduled-item">
```
Comment (ochiu) [inline auth-web/src/components/pay/eft/ShortNameAccountLink.vue:67]: After discussions with Ethan, additional states for alerts were removed due to changes in how they will be displayed in the refund flow. They will be implemented as part of refund flow.

File: auth-web/src/components/pay/eft/ShortNameAccountLink.vue [hunk 2]
```diff
@@ -216,6 +259,14 @@ export default defineComponent({
       emit('on-link-account', account)
     }
 
+    async function unlinkAccount (item) {
```
Comment (ochiu) [inline auth-web/src/components/pay/eft/ShortNameAccountLink.vue:262]: Added in basic unlinking to make things easier to test. 

Upcoming unlinking tickets will add in error handling / modals

**Comment by ochiu** [comment]
short-name-link-additional-info

File: auth-web/src/components/pay/eft/ShortNameAccountLink.vue [hunk 3]
```diff
@@ -57,50 +58,79 @@
...
-        <template #item-slot-actions="{ index }">
+        <template #expanded-item="{ item }">
+          <tr class="expanded-item-row">
+            <td
+              :colspan="headers.length"
+              class="pb-5 pl-0"
+            >
+              <span class="expanded-item scheduled-item">
+                <v-icon>mdi-clock-outline</v-icon>
+                {{ formatCurrency(item.amountOwing) }} will be applied to this account today at 6:00 p.m. Pacific Time.
```
Comment (seeker25) [inline auth-web/src/components/pay/eft/ShortNameAccountLink.vue:69]: hmm I think this is UTC, probably shifts with the time unfortunately with the CRON

File: auth-web/src/components/pay/eft/ShortNameAccountLink.vue [hunk 4]
```diff
@@ -57,50 +58,79 @@
...
-              Cancel Payment
-            </v-btn>
-            <span class="more-actions">
-              <v-menu
-                v-model="actionDropdown[index]"
-                :attach="`#action-menu-${index}`"
-                offset-y
-                nudge-left="74"
+            <!-- Cancel Payment / Unlink buttons-->
+            <template v-if="item?.hasPendingPayment">
```
Comment (seeker25) [inline auth-web/src/components/pay/eft/ShortNameAccountLink.vue:80]: this sucks for code highlighting haha

File: auth-web/src/components/pay/eft/ShortNameAccountLink.vue [hunk 5]
```diff
@@ -203,6 +235,17 @@ export default defineComponent({
       return `${props.shortNameDetails.accountId} ${props.shortNameDetails.accountName}`
     })
 
+    async function evaluateLinks () {
+      let pending = []
+      state.results.map(result => {
+        if (result.statusCode === 'PENDING' && result.amountOwing > 0) {
```
Comment (seeker25) [inline auth-web/src/components/pay/eft/ShortNameAccountLink.vue:241]: ENUM

Comment (seeker25) [reply auth-web/src/components/pay/eft/ShortNameAccountLink.vue:241]:
```
let pending = state.results.filter(result => 
  result.statusCode === 'PENDING' && result.amountOwing > 0
);

await Vue.nextTick();
state.expanded = [...pending];
```

File: auth-web/src/components/pay/eft/ShortNameAccountLink.vue [hunk 6]
```diff
@@ -216,6 +259,14 @@ export default defineComponent({
       emit('on-link-account', account)
     }
 
+    async function unlinkAccount (item) {
+      const response = await PaymentService.deleteShortNameLink(props.shortNameDetails.id, item.id)
+      if (!response) {
+        throw new Error('No response from delete short name link')
```
Comment (seeker25) [inline auth-web/src/components/pay/eft/ShortNameAccountLink.vue:265]: are these handled somehow?

File: auth-web/src/components/pay/eft/ShortNameAccountLink.vue (hunk 3 above)
Comment (seeker25) [reply auth-web/src/components/pay/eft/ShortNameAccountLink.vue:69]: `0 2 * * * default`

Comment (seeker25) [reply auth-web/src/components/pay/eft/ShortNameAccountLink.vue:69]: image

File: auth-web/src/components/pay/eft/ShortNameAccountLink.vue [hunk 7]
```diff
@@ -293,6 +346,41 @@ export default defineComponent({
...
+        justify-content: left;
+        grid-column: 1;
+        padding-right: 4px !important;
+        margin-right: 0px !important;
+      }
+    }
+
+    .alert-item {
+      .v-icon{
+        color: #F8661A;
```
Comment (seeker25) [inline auth-web/src/components/pay/eft/ShortNameAccountLink.vue:380]: Could we add this to the colors enum? or colors file?

File: auth-web/src/components/pay/eft/ShortNameAccountLink.vue (hunk 3 above)
Comment (seeker25) [reply auth-web/src/components/pay/eft/ShortNameAccountLink.vue:69]: it might be for example 6pm PST then 5 pm PDT

**Review by seeker25 — APPROVED** [review]

File: auth-web/src/components/pay/eft/ShortNameAccountLink.vue (hunk 3 above)
Comment (ochiu) [reply auth-web/src/components/pay/eft/ShortNameAccountLink.vue]: Checked with Ethan, added in a text change to mention this

File: auth-web/src/components/pay/eft/ShortNameAccountLink.vue (hunk 5 above)
Comment (ochiu) [reply auth-web/src/components/pay/eft/ShortNameAccountLink.vue]: ENUM and code update, thanks much cleaner

File: auth-web/src/components/pay/eft/ShortNameAccountLink.vue (hunk 4 above)
Comment (ochiu) [reply auth-web/src/components/pay/eft/ShortNameAccountLink.vue]: cleaned it up, conditional not needed

**Pull request merged**